### PR TITLE
[CALCITE-2722] Rewrite createLeftCall in iterative way instead of recursive way

### PR DESCRIPTION
**Pull request by hsyuan**
This is intended to avoid potential stack overflow caused by recursive function
call. Codes changes are covered by test case JdbcAdapterTest.testInPlan.

JIRA: https://issues.apache.org/jira/browse/CALCITE-2722

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java [hunk 1]
```diff
@@ -751,10 +750,14 @@ private SqlNode createLeftCall(SqlOperator op, List<SqlNode> nodeList) {
       if (nodeList.size() == 2) {
```
Comment (kgyrtkirk) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java]: I think we may also drop this "== 2" specialization as the method is no longer recursive
+1

Comment (hsyuan) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java]: updated, thanks!

File: core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java [hunk 2]
```diff
@@ -748,13 +747,14 @@ private SqlNode createSqlWindowBound(RexWindowBound rexWindowBound) {
     }
 
     private SqlNode createLeftCall(SqlOperator op, List<SqlNode> nodeList) {
-      if (nodeList.size() == 2) {
-        return op.createCall(new SqlNodeList(nodeList, POS));
+      assert nodeList != null && nodeList.size() >= 2;
```
Comment (kgyrtkirk) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java]: just some minor notes; code is fine.

* the summary of the commit headline should reflect more what the actual issue was and not describe the actual fix 
 - there are some guidelines here: https://calcite.apache.org/develop/ I would peek into `git log --grep Exception` to read a few examples
* I feel that this comment should really be in the body of the commit message
* this assert (if it fires) will just tell the user that something went wrong. You can either add a message or remove it as it seems like it works like this "by design"

Comment (hsyuan) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java]: Just notice your comments, sorry for being late.
Thanks for your review, I will take care for future PRs.

**Pull request closed without merging**